src/services/outliningElementsCollector.ts

```diff
@@ -6,9 +6,14 @@ namespace ts.OutliningElementsCollector {
     export function collectElements(sourceFile: SourceFile, cancellationToken: CancellationToken): OutliningSpan[] {
         const elements: OutliningSpan[] = [];
         let depth = 0;
+        const regions: RegionRange[] = [];
+        const regionText = "#region";
+        const regionStart = new RegExp("^\\s*//\\s*#region(\\s+.*)?$", "gm");
+        const regionEnd = new RegExp("^\\s*//\\s*#endregion(\\s|$)", "gm");

```

Comment:
Ah, I missed that this was a substring match (i.e. that the end of the line was open-ended).